components/popup/src/vwc-popup-base.ts

```diff
@@ -98,23 +121,34 @@ export class VWCPopupBase extends LitElement {
 
 	override connectedCallback(): void {
 		super.connectedCallback();
-		document.addEventListener('scroll', this.updatePosition);
+		window.addEventListener('scroll', this.updatePosition);
 		window.addEventListener('resize', this.onResizeWindow);
 	}
 
 	override disconnectedCallback(): void {
 		super.disconnectedCallback();
-		document.removeEventListener('scroll', this.updatePosition);
+		window.removeEventListener('scroll', this.updatePosition);
 		window.removeEventListener('resize', this.onResizeWindow);
 	}
 
-	protected override firstUpdated(changedProperties: PropertyValues): void {
-		super.firstUpdated(changedProperties);
-		this.updatePosition();
+	protected override firstUpdated(_changedProperties: PropertyValues): void {
+		super.firstUpdated(_changedProperties);
+		// Save the initial display state so that it can be restored when the positioning is complete
+		this.initialDisplayState = this.open;
+		this.hide();
+		this.anchorEl = this.getAnchorById();
+		// For proper positioning, show the popup after a delay when first updated
+		setTimeout(() => {
```

Comment:
Now I see why you have this delay.
It should be documented in the tests:
`it("should delay first position by 300ms")`
Or something like that. How did you get to 300ms BTW?

Reply:
@YonatanKra Having to set timeout, I chose 300 milliseconds because I saw FAST had a 300 millisecond delay for their tooltip to appear. 
Would you suggest a time of less than 300 milliseconds?

Reply:
As I said, we can go with 300ms, it just needs to be documented (in tests) as suggested. Why not 100ms (which is the maximum time for user interaction response)? Would it work then?
I guess 300ms is a heuristic because other factors might delay the page layout calculation (like long running JS or longer running calculations before the component loaded). 
So 100ms will not be enough.
By the way, we'd might want to add our own "ready" promise on this component and then use it in the tests. This might be a handy API.
I just don't feel good enough with arbitrary numbers going around.
In either case (arbitrary or not), we need to document this in the tests as this is part of the API exposed to the user (expect the component to finish rendering in 300ms).

Reply:
Thinking about it, I think we can "hide" this `setTimeout` behind an `updateComplete` message. Something like that in the `firstUpdated` function:
```
this.updateComplete = new Promise(res => setTimeout(() => {
      this.open = this.initialDisplayState;
			this.updatePosition();
			res();
}, this.DELAY));
```
I think that's the right syntax. Then the `updateComplete` will reflect the 300ms wait.

Reply:
This way, the `300ms` becomes an implementation details and not an unexpected side effect.

Reply:
@YonatanKra updateComplete is read-only

Reply:
Correct me if i'm wrong @rinaok but the denounce is required for document to be ready. The popup component has no way of knowing that

Reply:
Will this be of any help?

https://toruskit.com/blog/how-to-get-element-bounds-without-reflow/

Reply:
> @YonatanKra updateComplete is read-only

I did something with: `_getUpdateComplete`. Might help but seems an overkill.
If 300ms is a must then we need to document it in the tests (e.g. `should run positionUpdate on load after 300ms delay`).

Reply:
@rinaok if the intersection observer solves it, we may wanna create a single _**intersectionObserver**_ instance and reuse it.
notice it is being used as a service here
https://github.com/microsoft/fast/blob/master/packages/web-components/fast-foundation/src/utilities/intersection-service.ts